### PR TITLE
Move standard json codecs from soda-fountain

### PR DESCRIPTION
**Pull request by zachlefevre**

### CODE REVIEW

File: soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+            case (key, JString(s)) if (key == "latitude" || key == "longitude") =>
+              (
+                key,
+                try { JNumber(new java.math.BigDecimal(s)) }
+                catch { case _: NumberFormatException => JNull }
+              )
+            case (key @ "human_address", jo: JObject) =>
+              (key, JString(JsonUtil.renderJson(jo)))
+            case x => x
+          }.toMap
```
Comment (zachlefevre) [inline soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala:373]: This was modified from SF

File: soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+    def toJson(v: SoQLValue) =
+      v.typ.asInstanceOf[SoQLGeometryLike[T]].JsonRep(geometry(v))
+
+    def fromJValue(input: JValue) = {
+      input match {
+        case JNull        => Some(SoQLNull)
+        case JString(str) => fromWkt(str).map(geometry => value(geometry))
+        case _ => {
+          val geometry =
+            try { geoCodec.decode(input).toOption }
```
Comment (zachlefevre) [inline soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala:487]: as was this

File: soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+object JsonColumnRep {
+  // this is used to en/decrypt row IDs and values.  The key DOESN'T MATTER,
+  // because this system doesn't care about the actual values in a SoQLID or
+  // a SoQLValue; it just cares that it can recognize and reproduce them.
+  //
+  // It would be better to not care about the decrypted values at all, but alas
+  // that is not how SoQLID and SoQLVersion work.
+  private[this] val cryptProvider = new CryptProvider(Array[Byte](0))
+
+  val IdClearNumberRep = new SoQLID.ClearNumberRep(cryptProvider)
```
Comment (rjmac) [inline soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala:55]: ClearNumberRep likewise isn't a thing that something on the non-SF path should care about

File: soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+    }
+
+    def toJValue(value: SoQLValue): JValue =
+      if (SoQLNull == value) JNull
+      else JString(value.asInstanceOf[SoQLText].value)
+
+    val representedType: SoQLType = SoQLText
+  }
+
+  object ClientNumberRep extends JsonColumnRep {
```
Comment (rjmac) [inline soql-stdlib-codecs/src/main/scala/com/socrata/JsonColumnRep.scala:314]: Ditto this

**Pull request closed without merging**